### DISCUSSION TITLE
Deluge execute command cross seed

### DISCUSSION THREAD

**Discussion by Jerrk** [cross-seed/cross-seed#231] [Q&A] [answered]
I'm trying to use @afraazali's script from #164 that automatically searches for a cross seedable torrent when it gets successfully imported to sonarr/radarr. 
As mentioned in the topic a much better solution would be to use the torrent client "on download completion" function instead of the script getting called multiple times when importing a batch of files.
```
curl -XPOST http://192.168.1.111:2468/api/webhook \
  --data-urlencode 'infoHash='"${sonarr_download_id}"'' \
  --data-urlencode 'outputDir=/Sonarr'
```

admittedly im out of my depth here and i'd hope for some help in trying to figure out how to convert this so deluge can make sense of it.

```
curl -XPOST http://192.168.1.111:2468/api/webhook --data-urlencode 'infoHash='"${info_hash}"'' --data-urlencode 'outputDir='"${download_location}"''
```
does this look at all correct? 
i'm trying to find the deluge variables for the hash and download location to feed into cross-seed but i'm not sure of the formatting and/or the variables are correct
https://deluge.readthedocs.io/en/latest/reference/webapi.html

**Comment by mmgoodnow** [comment, marked as answer]
I found this link: https://dev.deluge-torrent.org/wiki/Plugins/Execute

It describes a pretty good procedure for setting up your pipeline incrementally.

Basically though, it seems like you just give it the path to an executable, and it will execute that file with the following three command line args: `"TorrentID" "Torrent Name" "Torrent Path"`.  I'm not sure what "TorrentID" is but I think it's probably infohash. 

From inside a bash script, you can reference these arguments as `$1`, `$2`, and `$3`, respectively. The starter bash script in the link would be an easy way to get a sense of what these values actually are. Make sure deluge's user has permission to execute the script. 

Once you've got qBittorrent successfully executing your script, then you can use some of the inputs recorded at /tmp/execute_script.log to execute your script manually, for testing the cross-seed end. 

Then, you can build up your script that way. Eventually it'll probably look something like 
```bash
#!/bin/bash
curl -XPOST http://192.168.1.111:2468/api/webhook \
  --data-urlencode "name=$2"
```

you can do it by name, or infohash, or both. If you want to pass outputDir, it'll look like `--data-urlencode "outputDir=/path/to/your/output/dir"`

**Reply by Jerrk** [reply]
took a bit of reading and testing to understand what was going but now it works 😁 

```
#!/bin/bash
torrentid=$1
torrentname=$2
curl -XPOST http://192.168.1.111:2468/api/webhook \
    --data-urlencode "infoHash=$torrentid" \
    --data-urlencode "name=$torrentname" \
    --data-urlencode "outputDir=/path/to/watch-folder/"
```
naming the variables is just to make it obvious whats what

**Reply by mmgoodnow** [reply]
Output dir seems wrong to me, but idk your workflow

**Reply by Jerrk** [reply]
ah yeah you are right that should be my watch-folder not the download location
:grin:
